frontend/src/components/core/EnterNickname.tsx

```diff
@@ -44,6 +44,9 @@ export default function EnterNicknamePage(props: EnterNicknameProps) {
    * with loading or error messages as necessary.
    */
   const enterNicknameActionUpdatePage = (nicknameParam: string): void => {
+    // Don't perform action if previous action is still running
+    if (loading) return;
+
```

Comment:
Similar to my other comment, can you change this to only run the function on `!loading`, and otherwise just not doing anything? This likely means you don't need a return type (even `void`?) for the function.